### PR TITLE
Added settings page

### PR DESCRIPTION
**Pull request by jankrzempek**
https://trello.com/c/3vxiFRbo/4-settings-for-profile-mvp-view

### CODE REVIEW

File: AirCasting/Settings/MyAccountView.swift [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+//
+
+import SwiftUI
+
+struct MyAccountView: View {
+    var body: some View {
+        ZStack {
+            VStack(alignment: .leading) {
+                
+                Text("You aren’t currently logged in")
```
Comment (mateuszmackowiak) [inline AirCasting/Settings/MyAccountView.swift:11]: in such cases where the logic isn't finished please add a TODO

Comment (mateuszmackowiak) [reply AirCasting/Settings/MyAccountView.swift:11]: Or this screen has an invalid name.

File: AirCasting/Settings/MyAccountViewSingOut.swift [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+//
+
+import SwiftUI
+
+struct MyAccountViewSingOut: View {
+    var body: some View {
+        ZStack {
+            VStack(alignment: .leading) {
+                
+                Text("You are currently logged in as jan.krzempek@lunarlogic.io")
```
Comment (mateuszmackowiak) [inline AirCasting/Settings/MyAccountViewSingOut.swift:11]: please add a todo.

File: AirCasting/Settings/SettingsView.swift [hunk 3]
```diff
@@ -10,26 +10,95 @@ import Foundation
...
-            
-            Spacer()
+            }
+            .listStyle(GroupedListStyle())
+            .navigationBarTitle("Settings")
+        }
+    }
+    
+    
+   private struct FullScreenModalView: View {
```
Comment (mateuszmackowiak) [inline AirCasting/Settings/SettingsView.swift:75]: Please think of a better name

File: AirCasting/Settings/SettingsView.swift [hunk 4]
```diff
@@ -10,23 +10,91 @@ import Foundation
 
 struct SettingsView: View {
     let logoutController: LogoutController
-
+    @State private var isToggle : Bool = false
```
Comment (pawelgil) [inline AirCasting/Settings/SettingsView.swift:13]: I know it's nit-picky, but the usual styling (in the app too afaict) is to put a colon after the name without any spaces:

```Swift
@State private var isToggle: Bool = false
```

File: AirCasting/Settings/MyAccountViewSingOut.swift [hunk 5]
```diff
@@ -0,0 +1,40 @@
+// Created by Lunar on 18/06/2021.
+//
+
+import SwiftUI
+
+struct MyAccountViewSingOut: View {
```
Comment (pawelgil) [inline AirCasting/Settings/MyAccountViewSingOut.swift:6]: typo `SingOut` -> `SignOut`

File: AirCasting/Settings/MyAccountView.swift [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+                }) {
+                    Group {
+                        HStack {
+                            Text("Create an account")
+                            Spacer()
+                            Image(systemName: "chevron.right")
+                        }
+                        .padding(.horizontal)
+                    }
+                } .buttonStyle(BlueButtonStyle())
```
Comment (pawelgil) [inline AirCasting/Settings/MyAccountView.swift:25]: remove space between `}` and `.` plz

File: AirCasting/Settings/MyAccountView.swift [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                            .stroke(lineWidth: 1.0)
+                    )
+                }
+                .padding()
+                Spacer()
+            }
+        }
+        .navigationTitle("My account")
+    }
+}
```
Comment (pawelgil) [inline AirCasting/Settings/MyAccountView.swift:50]: I think this body would be much more readable if we got rid of the unneeded details at the `body` level. [Top-level functions](https://stackoverflow.com/questions/633617/whats-the-difference-between-low-level-functions-top-level-functions) (computed var here is really a function with no paramaters, `var` is just Swift syntactic sugar) should really be screaming their intent, hiding any unneeded implementation detail in some private functions below. Anyone that wants to know the details would just scroll down to the part they are interested in and read it and everyone elses time is saved by making this high-level concept really easy to read thru. An example how this particular code can be divided:

```Swift
var body: some View {
    ZStack {
        VStack(alignment: .leading) {
            #warning("Logic here needs to be finished - handle log in")
            loggedOutInformationLabel
            createAccountButton
            logInButton
            Spacer()
        }
    }
    .navigationTitle("My account")
}

private var loggedOutInformationLabel: some View {
    Text("You aren’t currently logged in")
        .foregroundColor(.aircastingGray)
        .padding()
}

private var createAccountButton: some View {
    Button(action: {
    }) {
        Group {
            HStack {
                Text("Create an account")
                Spacer()
                Image(systemName: "chevron.right")
            }
            .padding(.horizontal)
        }
    }
    .buttonStyle(BlueButtonStyle())
    .padding()
}

private var logInButton: some View {
    Button(action: {
    }) {
        Group {
            HStack {
                Text("Login in")
                Spacer()
                Image(systemName: "chevron.right")
            }
            .padding(.horizontal)
            .padding(.vertical, 12)
        }
        .overlay(
            RoundedRectangle(cornerRadius: 5)
                .stroke(lineWidth: 1.0)
        )
    }
    .padding()
}
```

This simple little division makes it super-clear how this view looks like at just first glance.

File: AirCasting/Settings/MyAccountViewSingOut.swift [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    var body: some View {
```
Comment (pawelgil) [inline AirCasting/Settings/MyAccountViewSingOut.swift:7]: here the `body` is less complicated, but IMO would still benefit from making same thing as above (your call)

File: AirCasting/Settings/SettingsView.swift [hunk 9]
```diff
@@ -10,23 +10,91 @@ import Foundation
...
+    @State private var showModal = false
     var body: some View {
```
Comment (pawelgil) [inline AirCasting/Settings/SettingsView.swift:15]: this is a huge body function over here, let's try to cut it into smaller pieces like in the example above and to make it more readable this way

**Review by pawelgil — CHANGES_REQUESTED** [review]
Lets make it a bit cleaner 🧹
\ also minor typos

File: AirCasting/Settings/MyAccountView.swift [hunk 10]
```diff
@@ -8,47 +8,58 @@ struct MyAccountView: View {
...
                 Spacer()
             }
         }
-        .navigationTitle("My account")
+        .navigationTitle(Strings.MyAccountSettings.title)
     }
 }
 
+private var loggedOutInformationLabel: some View {
+    Text("You aren’t currently logged in")
```
Comment (pawelgil) [inline AirCasting/Settings/MyAccountView.swift:22]: missed this one, can go to `Strings` struct also :)

File: AirCasting/Settings/SettingsView.swift [hunk 11]
```diff
@@ -9,92 +9,136 @@ import SwiftUI
...
-                                    .accentColor(.black)
-                                Spacer()
-                                Image(systemName: "control")
-                                    .accentColor(.gray).opacity(0.6)
-                            }
-                        }
-                    }.sheet(isPresented: $showModal, content: {
-                        BackendSettingsModalView()
-                    })
+                    showModalButton
```
Comment (pawelgil) [inline AirCasting/Settings/SettingsView.swift:33]: good job on dividing stuff over here. One thing that can be better is this name. "showModalButton" is not really carrying any intent description. 'Modal' is just a presentation style, so showModalButton is saying nothing. Maybe lets think of a better name (from what I see here this button navigates the user to some backend settings, so maybe something like `navigateToBackendSettingsButton` ? )

File: AirCasting/Settings/SettingsView.swift [hunk 12]
```diff
@@ -9,92 +9,136 @@ import SwiftUI
...
                 Spacer()
-                Button("OK") {
-                    presentationMode.wrappedValue.dismiss()
-                }.buttonStyle(BlueButtonStyle())
-                
-                Button("Cancel") {
-                    presentationMode.wrappedValue.dismiss()
-                }.buttonStyle(BlueTextButtonStyle())
+                oKButton
+                cancelButton
```
Comment (pawelgil) [inline AirCasting/Settings/SettingsView.swift:121]: nice cleanup! 🧹

**Review by pawelgil — CHANGES_REQUESTED** [review]
great job on applying clean code techniques! Let's rename one more thing and cleanup missed string literal and we'll be good to go here

**Review by pawelgil — APPROVED** [review]
🔥

**Pull request merged**